packages/framework-core/src/booster-global-error-dispatcher.ts

```diff
@@ -78,20 +82,31 @@ export class BoosterGlobalErrorDispatcher {
   private async handleScheduleError(error: GlobalErrorContainer): Promise<Error | undefined> {
     if (!this.errorHandler || !this.errorHandler.onScheduledCommandHandlerError) throw error.originalError
     const currentError = error as ScheduleCommandGlobalError
-    return await this.errorHandler.onScheduledCommandHandlerError(currentError.originalError)
+    return await this.errorHandler.onScheduledCommandHandlerError(
+      currentError.originalError,
+      currentError.scheduleCommandEnvelope,
+      currentError.scheduleCommandMetadata
+    )
   }
 
   private async handleEventHandlerError(error: GlobalErrorContainer): Promise<Error | undefined> {
     if (!this.errorHandler || !this.errorHandler.onDispatchEventHandlerError) throw error.originalError
     const currentError = error as EventHandlerGlobalError
-    return await this.errorHandler.onDispatchEventHandlerError(currentError.originalError, currentError.eventInstance)
+    return await this.errorHandler.onDispatchEventHandlerError(
+      currentError.originalError,
+      currentError.eventEnvelope,
+      currentError.eventHandlerMetadata,
+      currentError.eventInstance
+    )
   }
 
-  private async handleReducerError(error: GlobalErrorContainer): Promise<Error | undefined> {
+  private async handleReducerError(error: GlobalErrorContainer): Promise<Error> {
     if (!this.errorHandler || !this.errorHandler.onReducerError) throw error.originalError
     const currentError = error as ReducerGlobalError
     return await this.errorHandler.onReducerError(
       currentError.originalError,
+      currentError.eventEnvelope,
+      currentError.reducerMetadata,
       currentError.eventInstance,
       currentError.snapshotInstance
```

Comment:
I wonder if we could just pass `currentError: ReducerGlobalError` as the only argument for this function, since all params are now being taken from the same object. On the other hand, this could make tests or other parts of the code more difficult, so we need to think about it

Reply:
See [this comment](https://github.com/boostercloud/booster/pull/1394#discussion_r1706051878).